metadata-ingestion/src/datahub/ingestion/source/powerbi/powerbi.py

```diff
@@ -260,7 +314,63 @@ def to_datahub_dataset(
                 aspect_name=Constant.STATUS,
                 aspect=StatusClass(removed=False),
             )
-            dataset_mcps.extend([info_mcp, status_mcp])
+            if self.__config.extract_dataset_schema:
```

Comment:
lets have a separate function for this logic

Reply:
Please create a function `extract_dataset_schema` similar to `self.extract_lineage` and move schema generation logic into  `extract_dataset_schema`